python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/ecs.py

```diff
@@ -0,0 +1,219 @@
+from pprint import pformat
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, cast
+
+import boto3
+import botocore
+import dagster._check as check
+from dagster import PipesClient
+from dagster._annotations import experimental
+from dagster._core.definitions.resource_annotation import TreatAsResourceParam
+from dagster._core.errors import DagsterExecutionInterruptedError
+from dagster._core.execution.context.compute import OpExecutionContext
+from dagster._core.pipes.client import (
+    PipesClientCompletedInvocation,
+    PipesContextInjector,
+    PipesMessageReader,
+)
+from dagster._core.pipes.utils import PipesEnvContextInjector, open_pipes_session
+
+from dagster_aws.pipes.message_readers import PipesCloudWatchMessageReader
+
+if TYPE_CHECKING:
+    from mypy_boto3_ecs.client import ECSClient
+    from mypy_boto3_ecs.type_defs import RunTaskRequestRequestTypeDef
+
+
+@experimental
+class PipesECSClient(PipesClient, TreatAsResourceParam):
+    """A pipes client for running AWS ECS tasks.
+
+    Args:
+        client (Optional[boto3.client]): The boto ECS client used to launch the ECS task
+        context_injector (Optional[PipesContextInjector]): A context injector to use to inject
+            context into the ECS task. Defaults to :py:class:`PipesEnvContextInjector`.
+        message_reader (Optional[PipesMessageReader]): A message reader to use to read messages
+            from the ECS task. Defaults to :py:class:`PipesCloudWatchMessageReader`.
+        forward_termination (bool): Whether to cancel the ECS task when the Dagster process receives a termination signal.
+    """
+
+    def __init__(
+        self,
+        client: Optional[boto3.client] = None,
+        context_injector: Optional[PipesContextInjector] = None,
+        message_reader: Optional[PipesMessageReader] = None,
+        forward_termination: bool = True,
+    ):
+        self._client: "ECSClient" = client or boto3.client("ecs")
+        self._context_injector = context_injector or PipesEnvContextInjector()
+        self._message_reader = message_reader or PipesCloudWatchMessageReader()
+        self.forward_termination = check.bool_param(forward_termination, "forward_termination")
+
+    @classmethod
+    def _is_dagster_maintained(cls) -> bool:
+        return True
+
+    def run(
+        self,
+        *,
+        context: OpExecutionContext,
+        extras: Optional[Dict[str, Any]] = None,
+        params: "RunTaskRequestRequestTypeDef",
```

Comment:
Actually @danielgafni per your comment in the other PR I think we name `params` `ecs_run_task_parms` to make it obvious this is a passthrough.

Reply:
yep doing that